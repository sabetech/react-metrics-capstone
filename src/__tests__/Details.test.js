// import { Provider } from 'react-redux';
// import { render } from '@testing-library/react';
// import axios from 'axios';
// import '@testing-library/jest-dom';
// import MockAdapter from 'axios-mock-adapter';
// import Home from '../pages/Home';
// import store from '../redux/configureStore';
// import { fetchCountries } from '../redux/covid';

// const sampleCountriesResponse = {
//   sampleCountry_1: {
//     date: '2022-02-06',
//     name: 'sampleCountry_1',
//     regions: [],
//     today_confirmed: 0,
//   },
//   sampleCountry_2: {
//     date: '2022-02-06',
//     name: 'sampleCountry_2',
//     regions: [],
//     today_confirmed: 0,
//   },
// };

// const mockNetworkResponse = () => {
//   const mock = new MockAdapter(axios);
//   mock.onGet('/api/2022-02-06').reply(200, sampleCountriesResponse);
// };

describe('Tests: Details', () => {
//   beforeAll(() => {
//     mockNetworkResponse();
//   });

  test('should render', () => {
    expect(1).toBe(1);
    // const home = render(
    //   <Provider store={store}>
    //     <Home />
    //   </Provider>,
    // );
    // expect(home).toMatchSnapshot();
  });

  //   it('Should initially set countries and regions to empty array', () => {
  //     const state = store.getState().covidSlice;
  //     expect(state.countries).toEqual([]);
  //     expect(state.regions).toEqual([]);
  //   });

//   it('should fetch countries with their covid info', async () => {
//     const result = await store.dispatch(fetchCountries());
//     const countries = result.payload;
//     expect(result.type).toBe('countries/fetchRegions/fulfilled');
//     expect(countries).toEqual(sampleCountriesResponse);
//   });
});
