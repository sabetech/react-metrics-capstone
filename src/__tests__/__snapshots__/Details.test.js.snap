// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests: Homepage should render 1`] = `
<div
  className="container"
>
  <nav>
    <h3
      className="heading"
    >
      Country-Based Covid Stats
    </h3>
  </nav>
  <div
    className="sticky-header"
  >
    <div
      className="card-label"
    >
      <p>
        Date: undefined
      </p>
    </div>
    <div
      className="stats-by-country"
    >
      <p>
        STATS BY COUNTRY
      </p>
    </div>
  </div>
  <ul
    className="country-list"
  />
</div>
`;
